File: 11interfaces.ts
interface IAnimal {
  walk(): void;
  eat(): void;
  sleep(): void;
}

class Dog implements IAnimal {

  walk(): void {
    console.log('on four legs...')
  }

  eat(): void {
    console.log('meat...')
  }

  sleep(): void {
    console.log('all day long...')
  }
}

File: 11interfaces.py
class IAnimal:

  # é definida uma classe com seus métodos sem implementação
  # para que cada classe filha faça a sua própria implementação
  # e se não fizer, o programa não quebra, apenas não faz nada (nesse caso)
  def walk():
    pass

  def eat():
    pass

  def sleep():
    pass

class Dog(IAnimal):

  def walk():
    print('on four legs...')

  def eat():
    print('meat...')

  def sleep():
    print('all day long...')
